Modules/Tests/YosemiteTests/Storage/PersistedProductConversionsTests.swift

```diff
@@ -0,0 +1,203 @@
+import Foundation
+import Testing
+@testable import Yosemite
+
+struct PersistedProductConversionsTests {
+
+    @Test("PersistedProduct init(from:) maps all POSProduct fields")
+    func test_product_init_from_posProduct_maps_all_fields() throws {
+        // Given
+        let siteID: Int64 = 1
+        let productID: Int64 = 10
+        let images: [ProductImage] = [
+            ProductImage(imageID: 100,
+                         dateCreated: Date(timeIntervalSince1970: 1),
+                         dateModified: nil,
+                         src: "https://example.com/1.png",
+                         name: "img1",
+                         alt: "alt1"),
+            ProductImage(imageID: 101,
+                         dateCreated: Date(timeIntervalSince1970: 2),
+                         dateModified: Date(timeIntervalSince1970: 3),
+                         src: "https://example.com/2.png",
+                         name: "img2",
+                         alt: nil)
+        ]
+        let attributes: [ProductAttribute] = [
+            ProductAttribute(siteID: siteID, attributeID: 0, name: "Color", position: 0, visible: true, variation: true, options: ["Red", "Blue"]),
+            ProductAttribute(siteID: siteID, attributeID: 0, name: "Size", position: 1, visible: false, variation: false, options: ["S", "M"])
+        ]
```

Comment:
super nit: since these aren't being asserted on in this test case, maybe can pass empty arrays for simplicity.